pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -112,3 +112,59 @@ def zyz_decomposition(U, wire):
     phis, thetas, omegas = map(math.squeeze, [phis, thetas, omegas])
 
     return [qml.Rot(phis, thetas, omegas, wires=wire)]
+
+
+def xyx_decomposition(U, wire):
+    r"""Recover the decomposition of a single-qubit matrix :math:`U` in terms
+    of elementary operations, as a product of X and Y rotations in the form
+    :math:`e^{i\gamma} RX(\phi) RY(\theta) RX(\lambda)`.
+
+    Args:
+        U (tensor): A 2 x 2 unitary matrix.
+        wire (Union[Wires, Sequence[int] or int]): The wire on which to apply the operation.
+
+    Returns:
+        list[qml.Operation]: Returns a list of 4 gates, an ``Identity`` times
+        the global phase, then an ``RX``, an ``RY``and another ``RX`` gate,
+        which when applied in the order of appearance in the list is equivalent
+        to the unitary :math:`U`.
+
+    **Example**
+
+    >>> U = np.array([[-0.28829348-0.78829734j,  0.30364367+0.45085995j],
+                      [ 0.53396245-0.10177564j,  0.76279558-0.35024096j]])
+    >>> decomp = xyx_decomposition(U, 0)
+    >>> decomp
+    [(0.38469215914523336-0.9230449299422961j)*(Identity(wires=[0])),
+            RX(0.45246583660683803, wires=[0]),
+            RY(1.3974974118006183, wires=[0]),
+            RX(-1.7210192479534632, wires=[0])]
+    """
+
+    # Small number to add to denominators to avoid division by zero
+    EPS = 1e-64
+
+    # Choose gamma such that exp(-i*gamma)*U is special unitary (detU==1).
+    gamma = math.cast_like(math.angle(math.linalg.det(U)) / 2, 1j)
```

Comment:
You have a utility private function in the file that you can use to convert to su2.

Reply:
I didn't use it at first because that function didn't return the phase, and I didn't want to modify it. But I think its a better idea to use it and add to it the flag `return_global_phase`, unless you think otherwise.